docs/csharp/csharp-7.md

```diff
@@ -16,64 +16,539 @@ ms.assetid: fd41596d-d0c2-4816-b94d-c4d00a5d0243
+
+You may want to specify the type of the `out` variable for clarity,
+as shown above. However, the language does support using an implicitly
+typed local variable:
+
+[!code-csharp[OutVarVariableDeclarations](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#02_OutVarVariableDeclarations "Implicitly typed Out variable")]
+
+<!--
+Add a sample at RC that shows how if statements
+scope out variables.
+-->
+
+Previously, you would need to separate the declaration of the out variable
+and its initialization into two different statements:
+
+[!code-csharp[OutVariableOldStyle](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#03_OutVariableOldStyle "classic out variable declaration")]
+
+This small language change improves your productivity in a couple ways.
+First, the code is easier to read. You declare the out variable where you
+use it, not on another line above.
+Second, by declaring the `out` variable where it is used in a method call,
+you don't need to assign an initial value, and you can't accidentally use
+it before it is assigned.
+
+The scope of the variable declaration minimizes the chance for mis-using
+the variable as well.  The `out` variable is scoped to the enclosing block.
+By minimizing the scope, you avoid name collisions with other variables and
+you minimize the chance of misuse.
+
+The most common use for this feature will be the `TryParse` pattern. In this
+pattern, a method returns a `bool` indicating success or failure, and an
+`out` variable that provides the result, in cases when the method succeeds.
 
 ## Tuples
 
-### Named and unnamed Tuples
+The syntax used for classes and structs provides a rich syntax to explain
+your design intent. But sometimes that rich syntax requires extra
+work with minimal benefit. You may often write methods that need a simple
+structure containing more than one data element. These scenarios are
+why *Tuples* were added to C#. Tuples are lightweight data structures
+that contain multiple elements.  You can't define methods, you simply
+have public get/set properties (without any validation) for the
+data members on the Tuple.
+
+> [!NOTE]
+> Tuples were available before C# 7, but had many limitations. Most importantly,
+> the members of these tuples were named `Item1`, `Item2` and so on.
+
+You can create a tuple by assigning each member to a value:
+
+[!code-csharp[UnnamedTuple](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#04_UnnamedTuple "Unnamed tuple")]
+
+That assignment creates a tuple whose members are `Item1` and `Item2`,
+following the existing `Tuple` syntax.
+You can modify that assignment to create a tuple that provides semantic
+names to each of the members of the tuple:
+
+[!code-csharp[NamedTuple](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#05_NamedTuple "Named tuple")]
+
+> [!NOTE]
+> The new features in Tuples require the `System.ValueTuple` type. For Visual Studio 15
```

Comment:
I think that you can change this to @System.ValueTuple. Is that OK, @mairaw
Also, I'd change this to read, "The new tuples features..." I don't think that I would capitalize Tuples.